### PR TITLE
Add an IVsTestingExtension

### PR DESCRIPTION
**Pull request by nkolev92**
Copy my IVSTestingExtension from https://github.com/nkolev92/NuGet.Tools/tree/master/IVsTestingExtension. 

Given that this is a tool, I don't plan on addressing all the comments, it's more of an FYI :) 
If you see issues/feel free to fix them yourself. I'll be happy to help if needed. :) 

@kartheekp-ms This one should build and work out of the box :)

### CODE REVIEW

File: IVsTestingExtension/NuGet.Config [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <config>
+    <add key="globalPackagesFolder" value="gpf" />
```
Comment (zivkan) [inline IVsTestingExtension/NuGet.Config]: why?

File: IVsTestingExtension/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        SYNC_TASKRUN_UNAWAITED,
+        SYNC_TASKRUN_BLOCKING,
+    }
+}
+```
+
+If you want add you own method instead, you only need to change 1 file, and that's src\Tests\TestMethodProvider.
+The infrastructure autowires the Provider and calls the GetMethod call to get the method that you want to test.
+
+![Example](screenshot.png)
```
Comment (zivkan) [inline IVsTestingExtension/README.md]: I suggest cropping the picture and/or resizing VS to be much smaller and mostly show the relevant thing. When viewing the readme and the screenshot gets resized to fit the width, I had absolutely no idea why the screenshot was relevant.

File: IVsTestingExtension/NuGet.Config (hunk 1 above)
Comment (nkolev92) [reply IVsTestingExtension/NuGet.Config]: In my experience, this tool involves a lot of custom testing. 
Having the gpf isolated avoids getting too many xprivates inside the gpf. 

I'll remove it.

File: IVsTestingExtension/README.md (hunk 2 above)
Comment (nkolev92) [reply IVsTestingExtension/README.md]: Cropped it, leaving only the left corner.

**Pull request merged**